python/hsfs/core/vector_db_client.py

```diff
@@ -155,9 +155,14 @@ def read(self, fg_id, keys=None, pk=None, index_name=None, n=10):
         if keys:
             query = {
                 "query": {
-                    "match": self._rewrite_result_key(
-                        keys, self._fg_fg_col_vdb_col_map[fg_id]
-                    )
+                    "bool": {
+                        "must": [
+                            {"match": {key: value}}
+                            for key, value in self._rewrite_result_key(
+                                keys, self._fg_col_vdb_col_map[fg_id]
+                            ).items()
```

Comment:
the result from `_rewrite_result_key` is already a map, cant it be directly inserted without this for loop?

Reply:
also a unit test would be good.

Reply:
opensearch requires this format `[{"match": {key: value}}]`, so need to reformat here.